File: leetcode/python/392_is_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) < 1:
            return True
        sIndex: int = 0
        for c in t:
            if s[sIndex] == c:
                sIndex += 1
                if sIndex == len(s):
                    return True
        return False

s = Solution()
print(s.isSubsequence(s = "abc", t = "ahbgdc"))
print(s.isSubsequence(s = "axc", t = "ahbgdc"))

File: leetcode/rust/src/392_is_subsequence.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        if s.len() > t.len() {
            return false;
        }
        if s == "" {
            return true;
        }

        let mut s: Vec<char> = s.chars().collect();
        let mut t: Vec<char> = t.chars().collect();
        let mut i = 0;
        for c2 in t {
            let c1 = s[i];
            if c1 == c2 {
                i += 1;
            }

            if i == s.len() {
                break;
            }
        }

        false
    }
}